### DISCUSSION TITLE
2FA Code expiring

### DISCUSSION THREAD

**Discussion by Anas-1554** [dani-garcia/vaultwarden#3978] [Q&A]
### Subject of the issue
2FA Code expiring.

### Deployment environment
A docker container on Oracle VPS.

* vaultwarden version:

How the server was installed: Docker image

I am facing a really weird issue from 2 days, My 2FA code have like 15 sec left to use when it use that code. It say expired but when I wait for another code to get generated it works fine. It is not like only one website is having this issues most of them are having this issues any possible fix?

## 2 Question.
If I have already hosted the docker image how I can enable admin page without breaking anything.

**Comment by BlackDex** [comment]
Check the `/admin/diagnostics` page and you would probably see that the date/time is out-of-sync.

Your server is probably not in sync with NTP. It probably is 15 seconds out of sync.

**Comment by Anas-1554** [comment]
![image](https://github.com/dani-garcia/vaultwarden/assets/94738878/ae2ab645-8644-4543-abd1-2ed46a9e969a)

**Reply by BlackDex** [reply]
Then you either are using an old version of Vaultwarden, or you need to enable the admin page. Try `/admin` and see what happens.

**Comment by Anas-1554** [comment]
That was my secound question how to do that without breaking my vault warden.

**Reply by stefan0xC** [reply]
Please read the [configuration overview](https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview) and [how to enable the admin page](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page).
